File: M. Rivqi Al Varras_22004_source code unit IV/matriks.py
def main():
    print("Operasi Perkalian 2 Buah Matriks")
m, n = map(int, input("Matriks pertama berorde : ").strip().split())
m1, n1 = map(int, input("Matriks kedua berorde : ").strip().split())
M1 = [[0 for _ in range(n)] for _ in range(m)]
M2 = [[0 for _ in range(n1)] for _ in range(m1)]
M3 = [[0 for _ in range(n1)] for _ in range(m)]
if n != m1 :
    print ("Perkalian matriks tidak dapat dilakukan, karena tidak sesuai dengan aturan perkalian matriks")
for i in range(m) :
    for j in range(n):
        M1[i][j] = int(input(f"Masukan elemen M1[{i+1} {j+1}] = "))
for i in range(m1) :
    for j in range (n1) :
        M2[i][j] = int(input(f"Masukan elemen M2[{i+1} {j+1}] = "))
for i in range(m) :
    for j in range(n1) :
        for k in range(n1) :
            M3[i][j] += M1[i][k] * M2[k][j]
print("Hasil perkaliannya adalah : ")
for i in range(m) :
    for j in range(n1):
        print(f"M3[{i+1} {j+1}] = ", M3[i][j])
if __name__ == "__main__":
    main()

File: M. Rivqi Al Varras_22004_source code unit IV/matriks.c
#include <stdio.h>
int main(){
   int n, m, n1, m1, i, j, k;
   int M1[100][100];
   int M2[100][100];
   int M3[100][100];
   int clrscr();
   printf("Operasi Perkalian 2 Buah Matriks\n");
   printf("Matriks pertama berorde : ");
   scanf("%i, %i", &m, &n);
   printf("Matriks kedua berorde : ");
   scanf("%i, %i", &m1, &n1);
   //printf("Perkalian matriks tidak dapat dilakukan, karena tidak sesuai dengan aturan perkalian matriks \n");
   if (n==m1);
    for(i=0; i<m; i++){
       for(j=0; j<n; j++){
       printf("Masukan elemen M1[%i %i] = ",i+1,j+1);
       scanf("%i", &M1[i][j]);
       }
   }
   for(i=0; i<m1; i++){
       for(j=0; j<n1; j++){
           printf("Masukan elemen M2[%i %i] = ",i+1,j+1);
           scanf("%i", &M2[i][j]);
       }
   }
   for(i=0; i<m; i++){
        for(j=0; j<n1; j++){
        M3[i][j] = 0;
   for(k=0; k<m1; k++){
        M3[i][j]=M3[i][j]+(M1[i][j]*M2[k][j]);
     }
    }
   }
        printf("Hasil perkaliannya adalah : "); 
    for(i=0; i<m; i++){
    for(j=0; j<n1; j++)
   {
         printf("M3[%i %i] = %i \n", i+1,j+1, M3[i][j]);
     }
       if (n==m1);
    //printf("Perkalian matriks tidak dapat dilakukan, karena tidak sesuai dengan aturan perkalian matriks \n");
   }
   return 0;
}